components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/auth/service/AuthServiceRequestWrapper.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authentication.framework.model.auth.service;
+
+import org.wso2.carbon.identity.application.authentication.framework.AuthenticatorFlowStatus;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatorData;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.auth.service.AuthServiceConstants;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+/**
+ * A HttpServletRequest wrapper to be used in authentication service.
+ */
+public class AuthServiceRequestWrapper extends HttpServletRequestWrapper {
+
+    private Map<String, String[]> parameters = new HashMap<>();
+    private List<AuthenticatorData> authInitiationData = new ArrayList<>();
+
+    public AuthServiceRequestWrapper(HttpServletRequest request, Map<String, String[]> parameters) {
+
+        super(request);
+        this.parameters = parameters;
+        setSessionDataKey(parameters);
+        request.setAttribute(AuthServiceConstants.AUTH_SERVICE_AUTH_INITIATION_DATA, authInitiationData);
+        skipNonceCookieValidation();
```

Comment:
Nonce cookie validation skipping is added temporarily and this logic will be improved in another PR. The same protection added by the nonce cookie should be available in some manner